src/test/TxTests.cpp

```diff
@@ -41,48 +42,72 @@ namespace txtest
 {
 
 bool
-applyCheck(TransactionFramePtr tx, Application& app)
+applyCheck(TransactionFramePtr tx, Application& app, bool checkSeqNum)
 {
     app.getDatabase().clearPreparedStatementCache();
 
     LedgerDelta delta(app.getLedgerManager().getCurrentLedgerHeader(),
                       app.getDatabase());
 
-    auto txSet = std::make_shared<TxSetFrame>(
-        app.getLedgerManager().getLastClosedLedgerHeader().hash);
-    txSet->add(tx);
-
-    // TODO: maybe we should just close ledger with tx instead of checking all
-    // of
-    // that manually?
+    AccountEntry srcAccountBefore;
 
     bool check = tx->checkValid(app, 0);
     TransactionResult checkResult = tx->getResult();
 
     REQUIRE((!check || checkResult.result.code() == txSUCCESS));
 
-    bool doApply;
-    // valid transaction sets ensure that
+    // now, check what happens when simulating what happens during a ledger
+    // close and reconcile it with the return value of "apply" with the one from
+    // checkValid:
+    // * an invalid (as per isValid) tx is still invalid during apply (and the
+    // same way)
+    // * a valid tx can fail later
+    auto code = checkResult.result.code();
+    if (code != txNO_ACCOUNT)
     {
-        auto code = checkResult.result.code();
-        if (code != txNO_ACCOUNT && code != txBAD_SEQ && code != txBAD_AUTH)
-        {
-            tx->processFeeSeqNum(delta, app.getLedgerManager());
-            doApply = true;
-        }
-        else
+        auto acnt = loadAccount(tx->getSourceID(), app, true);
+        srcAccountBefore = acnt->getAccount();
+
+        // no account -> can't process the fee
+        tx->processFeeSeqNum(delta, app.getLedgerManager());
+
+        // verify that the fee got processed
+        auto added = delta.added();
+        REQUIRE(added.begin() == added.end());
+        auto deleted = delta.deleted();
+        REQUIRE(deleted.begin() == deleted.end());
+        auto modified = delta.modified();
+        REQUIRE(modified.begin() != modified.end());
+        int modifiedCount = 0;
+        for (auto m : modified)
         {
-            doApply = (code != txBAD_SEQ);
+            modifiedCount++;
+            REQUIRE(modifiedCount == 1);
```

Comment:
cant we just check before loop if modified.size() == 1?

Reply:
it's a fake iterator, it only implements `begin` and `end`